### PR TITLE
Remaining system file updates

### PR DESCRIPTION
**Pull request by netopyr**
This PR adds logic to update the fee schedule and permissions after an override.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by iwsimon — DISMISSED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java [hunk 1]
```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
...
-                        + "transactions.");
-            }
+        // Get the current schedule
+        var currentSchedule = schedules.currentFeeSchedule();
+        if (currentSchedule == null) {
+            // If there is no current schedule, then we default to the default schedule. Since the default
+            // schedule is completely empty, this will effectively disable the handling of any transactions,
+            // since we don't know what to charge for them.
+            logger.warn("Unable to parse current fee schedule, will default to an empty schedule, effectively"
+                    + "disabling all transactions.");
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:105]: "disabling all _paid_ transactions."
I assume unpaid (i.e. privileged) transactions are still enabled, so the situation can be remedied (e.g. by updating the fee schedule).

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java [hunk 2]
```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
...
-                        + "immediately");
-                this.currentScheduleExpirationSeconds = 0;
-            }
+        // Populate the map of HederaFunctionality -> FeeData for the current schedule
+        this.currentFeeDataMap = new HashMap<>();
+        if (currentSchedule.hasTransactionFeeSchedule()) {
+            populateFeeDataMap(currentFeeDataMap, currentSchedule.transactionFeeScheduleOrThrow());
+        } else {
+            logger.warn("The current fee schedule is missing transaction information, effectively disabling all"
+                    + "transactions.");
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:115]: "paid transactions"?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java [hunk 3]
```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
...
-                // still get used since we continue to use the next schedule even if the expiration time has passed.
-                logger.warn("Unable to parse next fee schedule, will default to the current fee schedule.");
-                nextFeeDataMap = new HashMap<>(currentFeeDataMap);
+        // Get the expiration time of the current schedule
+        if (currentSchedule.hasExpiryTime()) {
+            this.currentScheduleExpirationSeconds =
+                    currentSchedule.expiryTimeOrThrow().seconds();
+        } else {
+            // If we don't have an expiration time, then we default to 0, which will effectively expire the
+            // current schedule immediately. This is the safest option.
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:124]: It is a safer option, but does monoservice do this (and if not do we need to match monoservice for now)?  I thought monoservice defaulted to continuing to use the "next" schedule indefinitely if the expiration time was 0.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java [hunk 4]
```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
...
-                // Populate the map of HederaFunctionality -> FeeData for the current schedule
-                this.nextFeeDataMap = new HashMap<>();
-                if (nextSchedule.hasTransactionFeeSchedule()) {
-                    populateFeeDataMap(nextFeeDataMap, nextSchedule.transactionFeeScheduleOrThrow());
-                } else {
-                    logger.warn("The next fee schedule is missing transaction information, effectively disabling all"
-                            + "transactions once it becomes active.");
-                }
+                logger.warn("The next fee schedule is missing transaction information, effectively disabling all"
+                        + "transactions once it becomes active.");
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:145]: "paid transactions"?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleManager.java [hunk 5]
```diff
@@ -45,10 +50,9 @@ public class ThrottleManager {
     private static final ThrottleDefinitions DEFAULT_THROTTLE_DEFINITIONS = ThrottleDefinitions.DEFAULT;
     public static final Set<HederaFunctionality> expectedOps = ExpectedCustomThrottles.ACTIVE_OPS.stream()
             .map(protoOp -> HederaFunctionality.fromProtobufOrdinal(protoOp.getNumber()))
-            .collect(Collectors.toSet());
+            .collect(Collectors.toCollection(() -> EnumSet.noneOf(HederaFunctionality.class)));
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleManager.java:53]: Just a question.  How does providing the empty `EnumSet` change semantics from the `toSet` collector?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java [hunk 6]
```diff
@@ -98,49 +105,51 @@ public void handleTxBody(@NonNull final HederaState state, @NonNull final Transa
...
-                // Updating the multiplier source to use the new gas throttle
-                // values that are coming from the network properties
-                congestionMultipliers.resetExpectations();
-            } else if (fileNum == config.hapiPermissions()) {
-                logger.error("Update of HAPI permissions not implemented");
-            } else if (fileNum == config.throttleDefinitions()) {
-                throttleManager.update(FileUtilities.getFileContent(state, fileID));
-                backendThrottle.rebuildFor(throttleManager.throttleDefinitions());
-                frontendThrottle.rebuildFor(throttleManager.throttleDefinitions());
+        if (fileNum == config.feeSchedules()) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:115]: Is there is an issue to add address book and node details support at some point?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java [hunk 7]
```diff
@@ -98,49 +105,51 @@ public void handleTxBody(@NonNull final HederaState state, @NonNull final Transa
...
+            congestionMultipliers.resetExpectations();
+            return result;
         }
+        return SUCCESS;
+    }
+
+    private FileID createFileID(final long fileNum, @NonNull final Configuration configuration) {
+        final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+        return FileID.newBuilder()
+                .realmNum(hederaConfig.realm())
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:150]: If these are system files, do we want to constrain realm and shard to `0`?  I don't think any system files are actually allowed to vary between shards and realms.

**Review by jsync-swirlds — APPROVED** [review]
Well designed, as usual.
I do have some questions to help me in understanding what's being done here.

**Review by kimbor — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 8]
```diff
@@ -707,8 +715,8 @@ private void genesis(@NonNull final MerkleHederaState state) {
         // And now that the entire dependency graph has been initialized, and we have config, and all migration has
         // been completed, we are prepared to initialize in-memory data structures. These specifically are loaded
         // from information held in state (especially those in special files).
-        initializeFeeManager(state);
         initializeExchangeRateManager(state);
+        initializeFeeManager(state);
```
[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:719]: I don't think the order matters right now. But `feeManager` has a reference to the `exchangeRageManager`, thus, it felt safer to initialize `exchangeRateManager` first.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:105]: I think you are right. The code existed prior to this PR and I did not review it thoroughly.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java (hunk 3 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java:124]: Again, I am not that familiar with the code. I believe this is what happens here. The current schedule expires immediately and we use the next schedule until something changes.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleManager.java (hunk 5 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleManager.java:53]: `expectedOps` is a `HashSet` when using `Collectors.toSet()` and an `EnumSet` with the new code.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java (hunk 6 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:115]: We don't have to do anything with the `addressBook` or `nodeDetails`. I did not know this when I wrote the placeholders above.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java (hunk 7 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:150]: Sharding and realms are still too fuzzy for me to have an opinion. 🙂

**Pull request merged**